index.html

```diff
@@ -3,11 +3,11 @@
 	<head>
 		<meta charset="utf-8" />
 		<meta name="viewport" content="width=device-width" />
-		<script src="assets/js/system.js"></script>
```

Comment:
I got rid of SystemJS because it wasn't really needed. Let me know if you have a use case for it beyond what I could ascertain.

Reply:
@joeldenning much appreciate you helping out on this! This project is not our real app, but a representation of how we're managing child apps. I believe it's following a similar approach to what you're doing, where we have child apps independently managed and released and we pull them into the parent app via SystemJS.

Reply:
Oh cool! Yeah that's how we do it at Canopy. For that case, if you want to have a "common dependencies" bundle that is shared between apps, I would recommend using [systemjs-builder](https://github.com/systemjs/builder) or [`jspm bundle`](https://jspm.io/docs/production-workflows.html#creating-a-bundle). The CommonChunksPlugin I am using here is a webpack-specific thing that might become tricky when mixing with SystemJS.

Reply:
Hi @joeldenning I've been trying off and on over the last couple of weeks to get this working with Webpack. You're right, it's pretty tricky, and nothing has worked so far. I tried a few things, like excluding Angular core and its related modules (ie. common dependencies) out of the Webpack build, and when that didn't work I tried filtering them out from the SystemJS import. Didn't work either.

I then tried JSPM on your recommendation. Hopefully that yields better results. Unfortunately I'm new to JSPM and I haven't had a successful build yet with app2 and app3 in this repo. Do you have an example config.js that accomplishes the equivalent of CommonChunksPlugin?

Reply:
Here's what I would recommend:

- only use jspm for the common dependencies bundle. Don't use jspm to build the child applications. To do this, (1) `jspm install` the child applications and (2) `jspm bundle @angular/angular @angular/router` (and anything else) to create the common dependencies bundle. Jspm will automatically update the config.js file if you use `jspm install` so you should be able to create your own. If you need more specific help with that, let me know and I can walk you through it.
- use SystemJS for your single-spa loading functions: `singleSpa.declareChildApplication('app1', () => SystemJS.import('/path-to-webpack-child-application.js'), () => true);`
- The child applications can still be built with webpack, but have them [build to and AMD format](https://webpack.js.org/configuration/output/#output-librarytarget) that have [externals](https://webpack.js.org/configuration/externals/#externals) for angular
- Have a separate webpack config for each child application.

If you do all of that, each child application can be owned/maintained by separate dev teams and even independently deployed. See https://github.com/CanopyTax/single-spa/issues/123 for some recent discussion about that.

Reply:
@joeldenning I'll give this a try. Thanks for the gameplan!